watchgod/main.py

```diff
@@ -139,10 +132,8 @@ async def __anext__(self) -> 'FileChanges':
                     return changes
 
     async def run_in_executor(self, func: 'AnyCallable', *args: Any) -> Any:
-        return await self._loop.run_in_executor(self._executor, func, *args)
-
-    def __del__(self) -> None:
-        self._executor.shutdown()
+        async with self._thread_semaphore:
+            return await anyio.to_thread.run_sync(func, *args)
```

Comment:
Pass the capacity limiter directly to `run_sync()`, or otherwise it will still use the default limiter.